encoder-otel-brave/src/test/java/zipkin2/reporter/otel/brave/ITZipkinSenderTests.java

```diff
@@ -0,0 +1,223 @@
+        jaegerAllInOne.getHttpOtlpPort());
+
+    List<String> traceIds = testingScenario.exportedTraceIds();
+
+    Awaitility.await().untilAsserted(() -> {
+      then(traceIds).hasSize(EXPECTED_TRACE_SIZE);
+      thenAllTraceIdsPresentInBackend(testingScenario.queryUrl(), traceIds);
+    });
+  }
+
+  @AfterEach
+  void shutdown() {
+    testingScenario.close();
+  }
+
+  private static void thenAllTraceIdsPresentInBackend(String queryUrl, List<String> traceIds) {
+    OkHttpClient client = new Builder()
+        .build();
+
+    traceIds.forEach(traceId -> {
+      Request request = new Request.Builder().url(queryUrl + traceId).build();
+      try (Response response = client.newCall(request).execute()) {
+        then(response.isSuccessful()).isTrue();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    });
+  }
+
+  /**
+   * Sender: Brave OKHttp with OTLP proto over HTTP ; Receiver: Jaeger OTLP
+   */
+  class JaegerTestingScenario {
+
+    private final BraveTraceIdGenerator braveTraceIdGenerator;
+
+    JaegerTestingScenario(int otlpPort) {
+      this.braveTraceIdGenerator = new BraveTraceIdGenerator(otlpPort);
+    }
+
+    String queryUrl() {
+      return "http://localhost:" + jaegerAllInOne.getQueryPort() + "/api/traces/";
+    }
+
+    List<String> exportedTraceIds() throws Exception {
+      return braveTraceIdGenerator.traceIds();
+    }
+
+    public void close() {
+      braveTraceIdGenerator.close();
+    }
+  }
+
+  /**
+   * Actual testing logic that uses Brave to generate spans and send them to the backend.
+   */
+  static class BraveTraceIdGenerator {
+
+    private final BraveHttpSenderProvider braveHttpSenderProvider = new BraveHttpSenderProvider();
+
+    private final int port;
+
+    Tracing tracing;
+
+    BraveTraceIdGenerator(int port) {
+      this.port = port;
+    }
+
+    List<String> traceIds() throws Exception {
+      ThreadLocalCurrentTraceContext braveCurrentTraceContext = ThreadLocalCurrentTraceContext.newBuilder()
+          .build();
+      List<String> traceIds = new ArrayList<>();
+      tracing = Tracing.newBuilder()
+          .currentTraceContext(braveCurrentTraceContext)
+          .supportsJoin(false)
+          .traceId128Bit(true)
+          .sampler(Sampler.ALWAYS_SAMPLE)
+          .addSpanHandler(braveHttpSenderProvider.apply(port))
+          .localServiceName("my-service")
+          .build();
+      Tracer braveTracer = tracing.tracer();
+
+      for (int i = 0; i < EXPECTED_TRACE_SIZE; i++) {
+        Span span = braveTracer.nextSpan().name("foo " + i)
+            .tag("foo tag", "foo value")
+            .kind(Kind.CONSUMER)
+            .error(new RuntimeException("BOOOOOM!"))
+            .remoteServiceName("remote service")
+            .start();
+        try (SpanInScope scope = braveTracer.withSpanInScope(span)) {
+          String traceId = span.context().traceIdString();
+          log.info("Trace Id <" + traceId + ">");
```

Comment:
remove this as there's only one test and one trace ID as a result of the desciption

Reply:
This class is replaced by [ITOtelEncoderTest](https://github.com/openzipkin-contrib/zipkin-otel/pull/7/files#diff-f1614b313ae6e7ebe173556465333def6f893b421571522e9bb496a42ede83c3)